src/plugins/intel_cpu/src/cpu_memory.h

```diff
@@ -364,8 +353,90 @@ class Memory : public IMemory {
     }
 };
 
+class StringMemory : public IMemory {
+public:
+    using OvString = ov::element_type_traits<ov::element::string>::value_type;
+
+    class StringMemoryMngr : public IMemoryMngr {
+    public:
+        StringMemoryMngr() : m_data(nullptr, release) {}
+        OvString* getStringPtr() const noexcept;
+        void setExtStringBuff(OvString* ptr, size_t size);
+        size_t getStrLen() const noexcept;
+
+        void* getRawPtr() const noexcept override;
+        void setExtBuff(void* ptr, size_t size) override;
+        bool resize(size_t size) override;
+        bool hasExtBuffer() const noexcept override;
+
+    private:
+        bool m_use_external_storage = false;
+        size_t m_str_upper_bound = 0lu;
+        std::unique_ptr<OvString, void (*)(OvString *)> m_data;
+
+        static void release(OvString* ptr) {}
+        static void destroy(OvString* ptr);
+    };
+
+    using StringMemoryMngrPtr = std::shared_ptr<StringMemoryMngr>;
+
+    StringMemory(const dnnl::engine& engine, const MemoryDescPtr& desc, const OvString* data = nullptr);
+
+    StringMemory(const dnnl::engine& engine, const MemoryDesc& desc, const OvString* data = nullptr) : StringMemory(engine, desc.clone(), data) {}
+
+    StringMemory(const dnnl::engine& engine, const MemoryDescPtr& desc, const StringMemoryMngrPtr& manager)
+        : m_engine(engine), m_mem_desc(desc), m_manager(manager) {}
+
+    bool isAllocated() const noexcept override {
+       return true;
+    }
```

Comment:
I guess it's not always allocated. Could you please add checks similar to `Memory` class.

Reply:
Done